### PR TITLE
Adds pre-commit

### PR DESCRIPTION
**Pull request by Jbarget**
pre-commit runs: 
- `mix test`
- `mix compile` (is this necessary?)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+-   repo: local
+    hooks:
+    -   id: mix-test
+        name: 'elixir: mixing tests'
+        entry: mix test
```
Comment (nelsonic) [inline .pre-commit-config.yaml]: @Jbarget this looks good. but do we need to add a dependency for it to work?
Do we need to `install` it with `brew`? http://pre-commit.com/#install
or is there an Elixir module we can use?

Comment (Jbarget) [reply .pre-commit-config.yaml]: True, i need to update the readme with instructions for installing `pre-commit`

installing with brew was the quickest way for me to get up and running.

Found [this pre_commit_hook] hex module but seems like its under construction. I haven't given it a go but could be a valid alternative. Here is it's github: https://github.com/tyrchen/ex_pre_commit_hook

**Comment by Jbarget** [comment]
@des-des is going to spike trying to get [pre_commit_hook](https://www.hex.pm/packages/pre_commit_hook) working.

We'll update the readme depending on what the outcome is

**Comment by nelsonic** [comment]
@Jbarget @des-des  https://github.com/tyrchen/ex_pre_commit_hook looks _very_ much like "Work In Progress". don't expend much time on it. we can come back to it. focus on the functionality described in the issue for now. 👍

**Comment by Jbarget** [comment]
@nelsonic i've updated the readme with instructions for `pre-commit` module

**Review by nelsonic — APPROVED** [review]
@Jbarget nice one. thanks! 👍

**Comment by des-des** [comment]
:+1:

**Pull request merged**